src/Models/Quote.php

```diff
@@ -62,8 +64,9 @@ protected static function booting()
                         "total_excl_tax", quote_item.row_total,
                         "attributes", quote_item_option.value,
                         "type", quote_item.product_type
-                    QUERY) . '
-                )), "$.null__") AS items')
+                    QUERY
+                        . '"backorder_count", IF(IF(stock.use_config_backorders, ' . $configBackorder . ', stock.backorders) = 2, GREATEST(0, quote_item.qty - stock.qty), 0)'
```

Comment:
You'd think that running the magento indexer will set the `stock.backorders` value to whatever the config setting is when `use_config_backorders` is enabled.

Not the case. If you change this setting on a product the value will be set correctly, but if you change the global config setting they won't be. Hence, I have to do a gross query like this.